src/Compilers/CSharp/Test/Semantic/Semantics/DelegateTypeTests.cs

```diff
@@ -13979,5 +13948,196 @@ int32[] arg2
+                int MethodWithParams(params int[] xs) => xs.Length;
+                DelegateNoParams d7 = MethodWithParams;
+                DelegateWithParams d8 = MethodNoParams;
+                DelegateNoParams d9 = (params int[] xs) => xs.Length; // 3
+                DelegateWithParams d10 = (int[] xs) => xs.Length;
+
+                delegate int DelegateNoDefault(int x);
+                delegate int DelegateWithDefault(int x = 1);
+                delegate int DelegateNoParams(int[] xs);
+                delegate int DelegateWithParams(params int[] xs);
+                """;
+            CreateCompilation(source).VerifyDiagnostics(
+                // (6,29): warning CS9501: Parameter 1 has default value '1' in lambda and '<missing>' in the target delegate type.
+                // DelegateNoDefault d4 = (int x = 1) => x; // 1
+                Diagnostic(ErrorCode.WRN_OptionalParamValueMismatch, "x").WithArguments("1", "1", "<missing>").WithLocation(6, 29),
+                // (7,31): warning CS9501: Parameter 1 has default value '2' in lambda and '1' in the target delegate type.
+                // DelegateWithDefault d5 = (int x = 2) => x; // 2
+                Diagnostic(ErrorCode.WRN_OptionalParamValueMismatch, "x").WithArguments("1", "2", "1").WithLocation(7, 31),
+                // (14,37): warning CS9502: Parameter 1 has params modifier in lambda and not in target delegate type.
+                // DelegateNoParams d9 = (params int[] xs) => xs.Length; // 3
+                Diagnostic(ErrorCode.WRN_ParamsArrayInLambdaOnly, "xs").WithArguments("1").WithLocation(14, 37));
+        }
+
+        [Fact]
+        public void DefaultsParamsConversion_Mix()
+        {
+            var source = """
+                D d1 = (int a, int b = 2, params int[] c) => { }; // 1, 2
+                void M(int a, int b = 2, params int[] c) { }
+                D d2 = M;
+                delegate void D(int x, int y, int[] z);
+                """;
+            CreateCompilation(source).VerifyDiagnostics(
+                // (1,20): warning CS9501: Parameter 2 has default value '2' in lambda and '<missing>' in the target delegate type.
+                // D d1 = (int a, int b = 2, params int[] c) => { }; // 1, 2
+                Diagnostic(ErrorCode.WRN_OptionalParamValueMismatch, "b").WithArguments("2", "2", "<missing>").WithLocation(1, 20),
+                // (1,40): warning CS9502: Parameter 3 has params modifier in lambda and not in target delegate type.
+                // D d1 = (int a, int b = 2, params int[] c) => { }; // 1, 2
+                Diagnostic(ErrorCode.WRN_ParamsArrayInLambdaOnly, "c").WithArguments("3").WithLocation(1, 40),
+                // (1,41): error CS1737: Optional parameters must appear after all required parameters
+                // D d1 = (int a, int b = 2, params int[] c) => { }; // 1, 2
+                Diagnostic(ErrorCode.ERR_DefaultValueBeforeRequiredValue, ")").WithLocation(1, 41));
```

Comment:
This error should go away once https://github.com/dotnet/roslyn/pull/64960 is merged.